types/express-session/index.d.ts

```diff
@@ -53,12 +53,14 @@ declare global {
 
 declare function session(options?: session.SessionOptions): express.RequestHandler;
 
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+
 declare namespace session {
   interface SessionOptions {
     secret: string | string[];
     name?: string;
     store?: Store | MemoryStore;
-    cookie?: express.CookieOptions;
+    cookie?: Omit<express.CookieOptions, 'secure'> & { secure?: boolean | 'auto' };
```

Comment:
I think we can deal with the problem without introducing `Omit`. Express session is one of most popular packages. If we upgrade minimum typescript version, millions of servers should be affected and might cause a huge disaster.

My suggestion is like the below. Could you try this?

```ts
cookie?: express.CookieOptions & { secure?: boolean | 'auto' }
```

Reply:
Yes, the first thing I tried.
![image](https://user-images.githubusercontent.com/1300172/67692374-67731b00-f9e3-11e9-8dd8-11798e8f2361.png)

cannot add incompatible types to the type of the parent attribute.
See https://stackoverflow.com/q/41285211

Reply:
Okay, if so, I think we should write a new interface rather than using `express.CookieOptions`.

Reply:
If we upgrade the minimum TypeScript version, I don't see how we'll cause a huge disaster. The servers should be package-locked to a version of the types, and if they are on an older version of TypeScript, they shouldn't get these newer definitions. I actually think that it would be OK to use `Omit` as that appears the proper idiomatic way of doing this.

Reply:
@cjbarth @CatsMiaow But, in the document of express-session, it doesn't say anything explicitly about that you can use same options of `express.CookieOptions` for `cookie` prop.

Reply:
@cjbarth @CatsMiaow But, in the document of express-session, it doesn't say anything explicitly about that you can use same options of express.CookieOptions for cookie prop.

Reply:
So using `express.CookieOptions` might be wrong in the first place.

Reply:
`express` and `express-session` have dependencies on the same `cookie` module.

https://github.com/expressjs/express/blob/master/lib/response.js#L30
https://github.com/expressjs/session/blob/master/session/cookie.js#L14

Reply:
It doesn't matter. Express-session is wraping cookie. It doesn't let you touch options for cookie module directly.